mitosheet/mitosheet/public/v3/sheet_functions/number_functions.py

```diff
@@ -57,62 +58,65 @@ def AVG(*argv: Union[int, float, None, pd.Series, RollingRange]) -> NumberFuncti
 @cast_values_in_arg_to_type('number')
 def MAX(*argv: Union[int, float, None, pd.Series, RollingRange]) -> NumberFunctionReturnType:
     
-    result: Union[pd.Series, float, int] = -sys.maxsize - 1
-
-    def get_new_result(_result: Union[pd.Series, float, int], new_value: Union[pd.Series, float, int]) -> Union[pd.Series, float, int]:
-        if isinstance(_result, pd.Series):
-            if isinstance(new_value, pd.Series):
-                return pd.concat([_result, new_value], axis=1).max(axis=1)
-            else:
-                return _result.apply(lambda v: np.nanmax([v, new_value]))
-        else:
-            if isinstance(new_value, pd.Series):
-                return new_value.apply(lambda v: np.nanmax([v, _result]))
-            else:
-                return max(_result, new_value)
+    result = get_final_result_series_or_primitive(
+        -sys.maxsize - 1,
+        argv,
+        lambda df: df.sum().sum(),
+        lambda previous_value, new_value: max(previous_value, new_value),
+        lambda previous_series, new_series: pd.concat([previous_series, new_series], axis=1).max(axis=1)
+    )
 
-    for arg in argv:
-        if isinstance(arg, pd.DataFrame):
-            df_sum = arg.sum().sum()
-            result = get_new_result(result, df_sum)
+    # If we don't find any arguements, we default to 0 -- like Excel
+    kept_default_max_value = not isinstance(result, pd.Series) and result == (-sys.maxsize - 1)
```

Comment:
Do we need this first check `not isinstance(result, pd.Series)`?

Reply:
Yes - because pandas throws a warning. If you check equality with a series, you get a series of booleans - which isn't what we want!